core/types/calls.cc

```diff
@@ -962,6 +970,107 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
         implicitKwsplat = true;
     }
 
+    {
+        const ArgInfo &probablyFirstKwarg = *pit;
+        if (gs.ruby3KeywordArgs && args.args.size() > 0 &&
+            // All positional arguments are already processed
+            ait == aPosEnd &&
+            // But not all arguments are processed, some keyword arguments remain
+            ait != aend &&
+            // First keyword parameter doesn't look like keyword param
+            !probablyFirstKwarg.flags.isKeyword && !probablyFirstKwarg.flags.isRepeated &&
+            !probablyFirstKwarg.flags.isBlock) {
+            auto isLastParam /* not counting last block param */ =
+                (pit - data->arguments.begin()) >= data->arguments.size() - 2;
+
+            auto areOtherParamsDefault = !isLastParam;
+            auto curIdx = pit - data->arguments.begin();
+
+            for (int i = data->arguments.size() - 1; i > curIdx; i--) {
+                auto flags = data->arguments[i].flags;
+                if (flags.isBlock || flags.isRepeated) {
+                    continue;
+                }
+                if (!flags.isDefault) {
+                    areOtherParamsDefault = false;
+                    break;
+                }
+            }
+
+            auto maybeKwargIdx =
+                ait - args.args.begin() < args.args.size() ? ait - args.args.begin() : args.args.size() - 1;
```

Comment:
It was originally a bound check, because I had a crash in `args.args[ ait - args.args.begin()]`. You're right, it's not needed anymore, I'll delete it